pkg/statistics/histogram.go

```diff
@@ -593,14 +593,16 @@ func (hg *Histogram) LessRowCount(sctx context.PlanContext, value types.Datum) f
 func (hg *Histogram) BetweenRowCount(sctx context.PlanContext, a, b types.Datum) float64 {
 	lessCountA := hg.LessRowCount(sctx, a)
 	lessCountB := hg.LessRowCount(sctx, b)
-	// If lessCountA is not less than lessCountB, it may be that they fall to the same bucket and we cannot estimate
-	// the fraction, so we use `totalCount / NDV` to estimate the row count, but the result should not greater than
-	// lessCountB or notNullCount-lessCountA.
-	if lessCountA >= lessCountB && hg.NDV > 0 {
+	rangeEst := lessCountB - lessCountA
+	aEqual, _ := hg.EqualRowCount(sctx, a, false)
+	bEqual, _ := hg.EqualRowCount(sctx, b, false)
```

Comment:
Not sure whether I understand this correctly. Can we use `bEqual` here? I see the comment of this function, it returns the number of row count where the column is LESS than value.

Reply:
This comment is encompassed by the next comment. I will resolve both comments with one solution as proposed below.